### DISCUSSION TITLE
biome-ignore useExhaustiveDependencies for specific variable

### DISCUSSION THREAD

**Discussion by valkum** [biomejs/biome#2509] [Ideas]
Currently, you can only disable the rule for the whole next line.
Often there is a valid case for `// biome-ignore useExhaustiveDependencies` it is due to some non-inferrable guarantees.
It would be nice if you can scope the ignore to a specific instance. This way, if the closure changes and would introduce a new missing dependency, the dev would get notified.

E.g. something like `// biome-ignore-instance lint/correctness/useExhaustiveDependencies: <VAR> <explanation>`

```ts
// biome-ignore-instance lint/correctness/useExhaustiveDependencies: contentRef comes from the useRef hook and thus is stable
useEffect(() => {
  some code that uses a var named contentRef. 
  This is a ref and thus stable but comes from something which doesn't fit in the stable config format.
}, []);

**Comment by arendjr** [comment]
Hm, I quite like this suggestion! It’s definitely a bummer when you have to disable *all* validation for a hook just because a single dependency needs an exception.

There is also a suggestion to split `useExhaustiveDependencies` into two rules (https://github.com/biomejs/biome/issues/630), to gain more granular control, but I’m starting to think the use case for that might (mostly?) go away if we implement this instead.

**Comment by arendjr** [comment]
This is now implemented and available in 1.8: https://biomejs.dev/linter/rules/use-exhaustive-dependencies/#ignoring-a-specific-dependency

**Reply by valkum** [reply]
Thank you very much!
